fix(config): relative cafile path works correctly

dereli opened the pull request:
**Summary**

When the cafile variable in the .yarnrc file is relative, the path to
file was resolved relative to current working directory not to the path
of the .yarnrc file.

Fixes: #6057

**Test plan**

Tests are added.

arcanis commented:
I unfortunately don't think this work. It will silently tolerate a directory settings (like `cache-folder`) to be a file or, even worse, not to exist at all (if one of its ancestors is a file). It would be better to have a `FOLDER_SETTINGS = ['cache-folder', ...]` whitelist that would decide whether to `mkdirp` or not.

dereli commented:
@arcanis Removed the try catch and introduced a new list of relative folder keys (`RELATIVE_KEYS_MAKE_DIR`), before mkdirp, we now check if the key is a folder.

Also, although the cases you mentioned didn't fail but not silently after update, I added tests for those two cases to make sure they always fail properly.

dereli commented:
@arcanis can you take a look here?
